dbms/src/IO/Compression/CompressionCodecMultiple.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <IO/Compression/ICompressionCodec.h>
+
+namespace DB
+{
+
+
+/** Compression codec for multiple codecs.
+  * It compresses data with multiple codecs and stores the list of codecs in the header.
+  * | <---------- Compression Header ---------> | <------------------------------------------- Additional Data -----------------------------------------------------------> | <- Compressed Data -> |
+  * | 0x91 - 1 byte | compressed size - 4 bytes | uncompressed size - 4 bytes | number of codecs - 1 byte | code 1 method byte - 1 byte | ... | code N method byte - 1 byte |   compressed block    |
```

Comment:
Can `uncompressed size` exceeds 4GiB? We use only 32bit here.

Reply:
```cpp
M(SettingUInt64, max_compress_block_size, DEFAULT_MAX_COMPRESS_BLOCK_SIZE, "The maximum size of blocks of uncompressed data before compressing for writing to a table.")

#define DEFAULT_MAX_COMPRESS_BLOCK_SIZE 1048576
```

The data to be compressed should be much less than 4GiB